core/common/src/main/java/alluxio/grpc/ServiceVersionClientServiceHandler.java

```diff
@@ -28,19 +32,32 @@
     extends ServiceVersionClientServiceGrpc.ServiceVersionClientServiceImplBase {
   /** Set of services that are going to be recognized by this versioning service. */
   private final Set<ServiceType> mServices;
+  @Nullable private final Supplier<NodeState> mNodeStateSupplier;
+  private final boolean mStandbyRpcEnabled =
+      Configuration.getBoolean(PropertyKey.STANDBY_MASTER_GRPC_ENABLED);
 
   /**
    * Creates service version handler that allows given services.
    * @param services services to allow
+   * @param nodeStateSupplier the supplier to get the node state
    */
-  public ServiceVersionClientServiceHandler(Set<ServiceType> services) {
+  public ServiceVersionClientServiceHandler(
+      Set<ServiceType> services, @Nullable Supplier<NodeState> nodeStateSupplier) {
     mServices = ImmutableSet.copyOf(Objects.requireNonNull(services, "services is null"));
+    mNodeStateSupplier = nodeStateSupplier;
   }
 
   @Override
   @SuppressFBWarnings(value = "DB_DUPLICATE_SWITCH_CLAUSES")
   public void getServiceVersion(GetServiceVersionPRequest request,
       StreamObserver<GetServiceVersionPResponse> responseObserver) {
+    if (mStandbyRpcEnabled
+        && mNodeStateSupplier != null && mNodeStateSupplier.get() == NodeState.STANDBY) {
```

Comment:
nit: what does `mNodeStateSupplier == null` imply?

Reply:
the grpc service is on worker